esp/bindings/http/platform/httpservice.cpp

```diff
@@ -215,6 +215,18 @@ EspHttpBinding* CEspHttpServer::getBinding()
     return thebinding;
 }
 
+//CORS allow headers for interoperability, we do not rely on this for security since
+//that only means treating the browser as a trusted entity.  We need to be diligent and secure
+//for every request whether it comes from a cross domain browser or any other source
+
+void checkSetCORSAllowOrigin(CHttpRequest *req, CHttpResponse *resp)
+{
+    StringBuffer origin;
+    req->getHeader("Origin", origin);
+    if (origin.length())
+        resp->setHeader("Access-Control-Allow-Origin", "*");
```

Comment:
Access-Control-Allow-Origin is effectively a response after validating access rights of the domain presented in the origin.  If we did want to add a way of restricting access to certain origin domains we would perform an access check here.  Similar to a form of IP restriction.